Sources/Private/Experimental/ExperimentalAnimationLayer.swift

```diff
@@ -155,21 +157,24 @@ extension ExperimentalAnimationLayer: RootAnimationLayer {
       animation.frameTime(forProgress: (presentation() ?? self).animationProgress)
     }
     set {
-      // Currently, setting `currentFrame` causes the existing layer hierarchy
-      // to be discarded and then completely rebuilt.
-      // This isn't a problem when setting the animation up just once,
-      // but can max out the CPU (causing frame drops / lag) for very-large animations
-      // when interactively controlling the current frame with a user gesture.
-      // TODO: This should be adjusted to use the existing layer hierarchy
-      //       and control something like `layer.timeOffset`.
-      setupAnimation(
-        context: .init(
-          playFrom: animation.startFrame,
-          playTo: animation.endFrame,
-          closure: nil),
-        timingConfiguration: .init(
-          speed: 0,
-          timeOffset: animation.time(forFrame: newValue)))
+      // We can display a specific frame of the animation by setting
+      // `timeOffset` of this layer. This requires setting up the layer hierarchy
+      // with a specific configuration (speed=0, etc) at least one. But if
+      // the layer hierarchy is already set up correctly, we can update the
+      // `timeOffset` very cheaply.
+      let requiredTimingConfiguration = CAMediaTimingConfiguration(speed: 0)
+
+      if
+        currentAnimationConfiguration?.timingConfiguration != requiredTimingConfiguration
+        || currentAnimationConfiguration?.animationContext.playFrom != animation.startFrame
+        || currentAnimationConfiguration?.animationContext.playTo != animation.endFrame
```

Comment:
nit: might be simpler to read if this logic is a method on `currentAnimationConfiguration`

Reply:
I'm tempted to make `AnimationConfiguration` equatable, but it contains a class that wraps a closure so I'm not entirely sure if it would be a good idea / idiomatic

Reply:
Ah, well that `AnimationCompletionDelegate` is already an `NSObject` subclass so it should already be `Equatable`. Maybe a non-issue then.